containers/java/Dockerfile

```diff
@@ -0,0 +1,6 @@
+FROM java:8u111-jdk
+
+RUN apt-get update && apt-get install -y \
+		                      gradle \
+                              git \
+        --no-install-recommends && rm -r /var/lib/apt/lists/*
```

Comment:
This container will be used as an executable task running on AWS ECS. To make it work nicely in this context, I think we need to:

- [ ] the wrapper script ( fetch_repo_and_collect_coverage.sh ) should be baked into the container
- [ ] we should define it as the ENTRYPOINT
- [ ] we should pass arguments to the container by using environment variables

Here is an example of this (have a look at the Dockerfile):
https://serverless.com/blog/serverless-application-for-long-running-process-fargate-lambda/